File: CSharp/AddBinary.cs
public class Solution {
    public string AddBinary(string a, string b) {
    	bool add1, add2 ;
    	char digSum ;
		string sum = "" ;
    	bool carry = false ;

		if ( a == null )
		{
			if ( b == null )
				sum = null ;
			else
				sum = b ;
		}
		else if ( a != null && b == null )
			sum = a ;
		else
		{
			int len = ( a.Length > b.Length ) ? a.Length : b.Length ;

			for ( int i = 0 ; i <= len ; i++ )
			{
				add1 = ( i < a.Length ) ? ( (a[a.Length-1-i] == '1') ? true : false ) : false ;
				add2 = ( i < b.Length ) ? ( (b[b.Length-1-i] == '1') ? true : false ) : false ;

				digSum = ( ( add1 ^ add2 ) ^ carry ) ? '1' : '0' ;
				carry = ( ( ( add1 ^ add2 ) & carry ) | ( add1 & add2 ) ) ? true : false ;

				sum = digSum + sum ;
			}
		}
		if ( sum[0] == '0' )
			sum = sum.Substring(1) ;
		return sum ;
    }

}

File: JavaScript/addBinary.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function(a, b) {

	var adder = ( x , y , c ) =>
	{
		var digSum = '0' ;
		if ( ( x ^ y ) ^ c )
			{ digSum = '1' ;}
		var carr = 0 ;
		if ( (x & y) | ( ( (x ^ y) &  c ) ) )
	 		{ carr = 1 ;}
		return { s:digSum , c:carr } ;
	};

	let sum = "" ;
  	let aL = a.length ;
  	let bL = b.length ;

	var diff = Math.abs( aL - bL ) ;
	var leadingZs = "" ;

	if ( diff > 0 )
		{ for ( diff ; diff > 0 ; diff -- )
			{ leadingZs += '0' ;} }
	if ( aL > bL )
		{ b = leadingZs + b ;}
	if ( bL > aL )
		{ a = leadingZs + a ;}

  	let len = a.length ;

  	var carry = 0 ;
	for ( var i = len - 1 ; i >= 0 ; i-- ) {
		var digSumCarry = adder ( ~~a[i] , ~~b[i] , carry ) ;
		sum = digSumCarry['s'] + sum ;
		carry = digSumCarry['c'] ;
	}

	if ( carry === 1 )
		{ sum = '1' + sum ;}

	return sum ;

};

/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
/*
var addBinary = function(a, b) {
    return (BigInt("0b"+a) + BigInt("0b"+b)).toString(2);
};
*/
